libraries/chain/combined_database.cpp

```diff
@@ -484,7 +504,7 @@ namespace eosio { namespace chain {
             if (backing_store == backing_store_type::ROCKSDB) {
                auto prefix_key = eosio::session::shared_bytes(rocksdb_contract_kv_prefix.data(),
                                                                    rocksdb_contract_kv_prefix.size());
-               auto key_values = std::vector<std::pair<eosio::session::shared_bytes, eosio::session::shared_bytes>>{};
+               auto key_values = std::deque<std::pair<eosio::session::shared_bytes, eosio::session::shared_bytes>>{};
```

Comment:
The decision was to have a batching size, so deque will not be more efficient.  My code writes the batch when it reaches a certain size.